src/application-events-model.ts

```diff
@@ -0,0 +1,365 @@
+import {
+  BumperEvents,
+  DownloadEvents,
+  DualscreenEvents,
+  NavigationEvents,
+  PlaylistEvents,
+  ShareEvents,
+  TranscriptEvents,
+  SkipEvents
+} from './new-applications-events';
+import { KavaModel } from './kava-model';
+import { KavaEvent } from './types';
+import { ButtonType } from './enums/button-type';
+import { KalturaApplication } from './enums/kaltura-application';
+import { ApplicationEventType } from './enums/application-event-type';
+import { PageLoadType } from './enums/page-load-type';
+
+export function getApplicationEventsModel(eventObj: KavaEvent, model: KavaModel, innerEventPayload: any): any {
+  const commonModel = {
+    partnerId: model.getPartnerId(),
+    entryId: model.getEntryId(),
+    sessionId: model.getSessionId(),
+    kalturaApplication: KalturaApplication.PLAYER
+  };
+  if (model.getUserId()) {
+    commonModel['userId'] = model.getUserId();
+  }
+  const eventModel = eventObj.getEventModel(innerEventPayload);
+  return Object.assign(eventModel, commonModel);
+}
+
+export const ApplicationEventsModel: { [playerEventName: string]: KavaEvent } = {
+  [DualscreenEvents.CHANGE_LAYOUT]: {
+    type: 'CHANGE_LAYOUT',
+    getEventModel: (payload: any): any => ({
+      eventType: ApplicationEventType.BUTTON_CLICKED,
+      buttonName: 'Dual__screen_change_layout',
+      buttonType: ButtonType.Unknown,
+      buttonValue: payload.layout
+    })
+  },
+  [ShareEvents.SHARE_CLICKED]: {
+    type: 'SHARE_CLICKED',
+    getEventModel: (): any => ({
+      eventType: ApplicationEventType.BUTTON_CLICKED,
+      buttonName: 'Share_embed_open',
```

Comment:
button names should be consts

Reply:
why

Reply:
Since there is no reuse for those names, and they exist only in one place, here, lets leave it as is